e2e/vuln_1783967524.py

```
@@ -0,0 +1,14 @@
+import hashlib, sqlite3
+
+def make_token(password):
+    # Weak hashing — MD5 is not suitable for secrets
+    return hashlib.md5(password.encode()).hexdigest()
+
+def get_user(db, user_id):
+    cur = db.cursor()
+    # SQL injection via string concatenation
+    cur.execute("SELECT * FROM users WHERE id = '" + user_id + "'")
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)